libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -648,7 +648,9 @@ ASTPointer<Mapping> ASTJsonImporter::createMapping(Json::Value const&  _node)
 	return createASTNode<Mapping>(
 		_node,
 		convertJsonToASTNode<TypeName>(member(_node, "keyType")),
-		convertJsonToASTNode<TypeName>(member(_node, "valueType"))
+		convertJsonToASTNode<TypeName>(member(_node, "valueType")),
+		convertJsonToASTNode<ASTString>(member(_node, "keyName")),
+		convertJsonToASTNode<ASTString>(member(_node, "valueName"))
```

Comment:
The equivalent code was missing in `ASTJsonExporter`. This is why the cli test was failing with
```
Looking at 2940 .sol files...
................................................................................................................................................ERROR: AST reimport failed for input file /home/hari/s/r2/test/libsolidity/syntaxTests/array/pop/storage_with_mapping_pop.sol

Compiler stderr:
Failed to import AST: JSON-Node needs to have 'nodeType' and 'id' fields.

Compiler stdout:

```
I'll fix this.

Kinda wish we had serde magic here :(

Reply:
There is also a nasty bug here. `ASTString` is not an ASTNode. So this conversion would fail at runtime.

The correct way to do it would be:
```diff
-		convertJsonToASTNode<ASTString>(member(_node, "keyName")),
-		convertJsonToASTNode<ASTString>(member(_node, "valueName"))
+		memberAsASTString(_node, "keyName"),
+		memberAsASTString(_node, "valueName")
```